L/LLVM/LLVM_full@12.0.0/build_tarballs.jl

```diff
@@ -3,4 +3,4 @@ version = v"12.0.0"
 include("../common.jl")
 
 build_tarballs(ARGS, configure_build(ARGS, version; experimental_platforms=true)...;
-                     preferred_gcc_version=v"7", preferred_llvm_version=v"8", julia_compat="1.7")
+                     preferred_gcc_version=v"12", preferred_llvm_version=v"8", julia_compat="1.7")
```

Comment:
There is no GCC 12 :thinking:

Reply:
I'm wondering if we should build a more recent GCC for this platform, we're using a [commit](https://github.com/iains/gcc-darwin-arm64/commit/ccc57f4ed3feed697f17d3230786389b1b410af9) from November

Reply:
Yeah I don't understand why now suddenly this is a problem.

Reply:
BTW, the offending function is the same as https://github.com/JuliaLang/julia/issues/40703

Reply:
Also we are using clang on MacOS anyway right?

Reply:
We should, yes

Reply:
> BTW, the offending function is the same as [JuliaLang/julia#40703](https://github.com/JuliaLang/julia/issues/40703)

So adding a dependency like `Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))` might help?

Reply:
`CompilerSupportLibraries_jll` is really useful only at runtime, those libraries should be already available during the build

Reply:
Well, I'm confused:
```console
% julia -e 'using BinaryBuilderBase; BinaryBuilderBase.runshell(Platform("aarch64", "macos"); preferred_gcc_version=v"11")'
sandbox:${WORKSPACE} # nm /opt/${target}/${target}/lib/libstdc++.${dlext} | grep throw_bad_array_new_length
00000000000fb340 S __ZSt28__throw_bad_array_new_lengthv
00000000000fb088 S ___cxa_throw_bad_array_new_length
sandbox:${WORKSPACE} # nm /opt/${target}/${target}/lib/libstdc++.${dlext} | cut -d' ' -f3 | c++filt --strip-underscore | grep throw_bad_array_new_length
std::__throw_bad_array_new_length()
___cxa_throw_bad_array_new_length
```
So the symbol is there?

Reply:
Of course, this confusion is besides the confusion about how a libstdc++ symbol is involved here

Reply:
It's not the target version... It's the bootstrap.

```
[09:57:03] Error relocating /workspace/bootstrap/bin/clang-tblgen: _ZSt28__throw_bad_array_new_lengthv: symbol not found
```
https://github.com/JuliaPackaging/Yggdrasil/blob/96a818e6ea1a16c337df9a70ccf6dda5016e96e2/L/LLVM/common.jl#L74-L87

Reply:
Uhm, what if you revert https://github.com/JuliaPackaging/Yggdrasil/commit/5dca45654c1d19458cb624df3c69e4fd1163dfb2 here?

Reply:
```
sandbox:${WORKSPACE} # /opt/x86_64-linux-musl/bin/x86_64-linux-musl-nm  /opt/x86_64-linux-musl/x86_64-linux-musl/lib64/libstdc++.so.6.0.29 |  grep throw_bad_array_new_length
00000000000c0829 T _ZSt28__throw_bad_array_new_lengthv
00000000000be176 T __cxa_throw_bad_array_new_length
```

also has it...